client/interfaces.go

```diff
@@ -277,6 +278,22 @@ type ContainerSnapshotCopyArgs struct {
 	Live bool
 }
 
+// The ContainerConsoleArgs struct is used to pass additional options during a
+// container console session
+type ContainerConsoleArgs struct {
+	// Standard input
+	Stdin io.ReadCloser
+
+	// Standard output
+	Stdout io.WriteCloser
+
+	// Standard error
+	Stderr io.WriteCloser
```

Comment:
Since we can't really support the partial attach that we do for Exec, shouldn't we just change that to just:

    Terminal io.ReadWriteCloser